app/connector/sql/src/main/java/io/syndesis/connector/sql/common/JSONBeanUtil.java

```diff
@@ -73,16 +73,30 @@ public static Properties parsePropertiesFromJSONBean(final String json) {
     }
 
     public static Map<String,SqlParameterValue> parseSqlParametersFromJSONBean(final String json, final Map<String, Integer> jdbcTypeMap) {
+        final Map<String,SqlParameterValue> ret = new HashMap<>();
+
+        if (json == null || json.length() == 0) {
+            return ret; // json is empty so no need to parse
+        }
+
         final Map<String, String> parsed;
         try {
             parsed = MAPPER.readerFor(STRING_STRING_MAP).readValue(json);
         } catch (final IOException e) {
             throw new IllegalArgumentException("Unable to parse given JSON", e);
         }
-        final Map<String,SqlParameterValue> ret = new HashMap<>();
         for (String key : parsed.keySet()) {
             Object value = parsed.get(key);
-            Integer jdbcType = (jdbcType = jdbcTypeMap.get(key)) != null ? jdbcType : Types.VARCHAR;
+
+            Integer jdbcType = null;
+            if (jdbcTypeMap != null) {
```

Comment:
looking at:

https://github.com/syndesisio/syndesis/blob/93d1784d2de1ee354e651325bace008cb27d0ba3/app/connector/sql/src/main/java/io/syndesis/connector/sql/customizer/SqlConnectorCustomizer.java#L82-L98

it doesn't seem that `jdbcTypeMap` can be `null`

Reply:
@zregvart 
You are right and did consider this. However, upon writing the unit test and the static nature of the function, it seemed prudent that since I was handling the possibility of 1 parameter being null, I might as well check the other. Probably unnecessary but at least consistent.